src/main/java/org/folio/circulation/resources/subscribers/LoanRelatedFeeFineClosedSubscriberResource.java

```diff
@@ -0,0 +1,115 @@
+  private static final List<String> LOST_ITEM_FEE_TYPES = Arrays.asList(
+    LOST_ITEM_FEE_TYPE, LOST_ITEM_PROCESSING_FEE_TYPE);
+
+  public LoanRelatedFeeFineClosedSubscriberResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    new RouteRegistration("/circulation/subscribers/loan-related-fee-fina-closed", router)
+      .create(this::handleFeeFineClosedEvent);
+  }
+
+  private void handleFeeFineClosedEvent(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+
+    createAndValidateRequest(routingContext)
+      .after(request -> processEvent(context, request))
+      .thenApply(r -> r.toFixedValue(NoContentResponse::noContent))
+      .thenAccept(context::writeResultToHttpResponse);
+  }
+
+  private CompletableFuture<Result<Loan>> processEvent(
+    WebContext context, LoanRelatedFeeFineClosedEvent event) {
+
+    final Clients clients = create(context, client);
+    final LoanRepository loanRepository = new LoanRepository(clients);
+
+    return loanRepository.getById(event.getLoanId())
+      .thenCompose(r -> r.after(loan -> {
+        if (loan.isDeclaredLost()) {
+          return closeDeclaredLostLoanIfLostFeesResolved(clients, loan);
+        }
+
+        return completedFuture(succeeded(loan));
+      }));
+  }
+
+  private CompletableFuture<Result<Loan>> closeDeclaredLostLoanIfLostFeesResolved(
+    Clients clients, Loan loan) {
+
+    final AccountRepository accountRepository = new AccountRepository(clients);
+    final LostItemPolicyRepository lostItemPolicyRepository = new LostItemPolicyRepository(clients);
+    final StoreLoanAndItem storeLoanAndItem = new StoreLoanAndItem(clients);
+
+    return accountRepository.findAccountsForLoan(loan)
+      .thenComposeAsync(lostItemPolicyRepository::findLostItemPolicyForLoan)
+      .thenCompose(r -> r.after(loanWithReferenceData -> {
+        if (areLostFeesClosed(loanWithReferenceData)) {
+          loanWithReferenceData.closeLoanAsLostAndPaid();
+
+          return storeLoanAndItem.updateLoanAndItemInStorage(loanWithReferenceData);
+        }
+
+        return completedFuture(succeeded(loanWithReferenceData));
+      }));
+  }
+
+  private boolean areLostFeesClosed(Loan loan) {
+    if (loan.getLostItemPolicy().hasActualCostFee()) {
+      // Actual cost fee is processed manually
+      return false;
+    }
+
+    return loan.getAccounts().stream()
+      .filter(account -> LOST_ITEM_FEE_TYPES.contains(account.getFeeFineType()))
+      .allMatch(Account::isClosed);
+  }
+
+  private Result<LoanRelatedFeeFineClosedEvent> createAndValidateRequest(RoutingContext context) {
+    final LoanRelatedFeeFineClosedEvent eventPayload = fromJson(context.getBodyAsJson());
+
+    if (eventPayload.getLoanId() == null) {
+      return failed(singleValidationError(
+        new ValidationError("Loan id is required", "loanId", null)));
+    }
```

Comment:
What will happen to this validation error?

Reply:
I guess, pub/sub will log this error in its internal audit table.

Reply:
Please could it be confirmed what pub-sub's reaction to this will be.